common/src/main/java/com/flipkart/varadhi/common/VaradhiCache.java

```diff
@@ -1,83 +1,208 @@
 package com.flipkart.varadhi.common;
 
-import com.flipkart.varadhi.common.exceptions.VaradhiException;
-import com.google.common.base.Ticker;
-import com.google.common.cache.Cache;
-import com.google.common.cache.CacheBuilder;
-import com.google.common.util.concurrent.UncheckedExecutionException;
 import io.micrometer.core.instrument.Counter;
 import io.micrometer.core.instrument.MeterRegistry;
 import io.micrometer.core.instrument.Tags;
+import lombok.extern.slf4j.Slf4j;
 
-import java.util.concurrent.ExecutionException;
-import java.util.function.BiFunction;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 import java.util.function.Function;
 
 /**
- * TODO: lets not put varadhi word in every other class name. VaradhiCache does not tell us its purpose.
+ * A thread-safe generic cache implementation with metrics support.
+ * This cache provides basic operations like get, put, invalidate with metrics tracking
+ * for hits, misses, and size.
  *
- * @param <K>
- * @param <V>
+ * @param <K> the type of keys maintained by this cache
+ * @param <V> the type of mapped values
  */
-public class VaradhiCache<K, V> {
-    // This is NonLoading cache. Get wll invoke loader() on calling thread when needed and hence possibly blocking in
-    // some cases.
-    // TODO:: need to support unblocking loader call.
-    private final Cache<K, V> entityCache;
-    private final Function<K, V> entityProvider;
-    private final Counter getCounter;
+@Slf4j
+public final class VaradhiCache<K, V> {
```

Comment:
lets just use https://github.com/ben-manes/caffeine?

Reply:
I think we can literally remove this class. I dont see a need for this class. If we need a cache then we can directly use caffiene